src/main/java/dk/cngroup/lentils/service/ImageService.java

```diff
@@ -0,0 +1,135 @@
+import dk.cngroup.lentils.exception.ImageFileNotFoundException;
+import dk.cngroup.lentils.exception.ImageNotSavedException;
+import dk.cngroup.lentils.exception.ResourceNotFoundException;
+import dk.cngroup.lentils.repository.ImageRepository;
+import org.apache.commons.fileupload.disk.DiskFileItem;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import javax.transaction.Transactional;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+@Service
+public class ImageService {
+
+    private static final String SEPARATOR = File.separator;
+    private static final String IMAGE_DIRECTORY = SEPARATOR + "upload" + SEPARATOR + "images" + SEPARATOR;
+
+//    @Value("${image-directory}")
+//    private String imageDirectory;
+
+    private final ImageRepository imageRepository;
+
+    public ImageService(final ImageRepository imageRepository) {
+        this.imageRepository = imageRepository;
+    }
+
+    public Image save(final Image image) {
+        return imageRepository.save(image);
+    }
+
+    public MultipartFile getFile(final String stringPath) {
+        final File file = new File(stringPath);
+        DiskFileItem fileItem = new DiskFileItem("file", "image",
+                true, file.getName(), (int) file.length(), file.getParentFile());
+        try {
+            fileItem.getOutputStream();
+        } catch (IOException e) {
+            throw new ImageFileNotFoundException();
+        }
+        MultipartFile result = new CommonsMultipartFile(fileItem);
+        return result;
+    }
+
+    public String getFileNameFromEntity(final Hint hint) {
+        return isFilePresentInHintEntity(hint) ? hint.getImage().getPath() : "";
+    }
+
+    public String getFileName(final Hint hint, final HintFormObject formObject) {
+        if (isFilePresentInHintForm(formObject)) {
+            return formObject.getImage().getOriginalFilename();
+        }
+            return getFileNameFromEntity(hint);
+    }
+
+    public void setImageToFormObject(final Hint hint, final HintFormObject formObject) {
+        if (isFilePresentInHintEntity(hint)
+                && (!isFilePresentInHintForm(formObject))) {
+            formObject.setImage(getFile(hint.getImage().getPath()));
+        }
+    }
+
+    public String getFileNamefromFormObject(final HintFormObject formObject) {
+        return isFilePresentInHintForm(formObject) ? formObject.getImage().getOriginalFilename() : "";
+    }
+
+    public String getFilePath(final MultipartFile file) {
+        if (file.isEmpty()) {
+            throw new ImageFileNotFoundException();
+        }
+        String fileName = file.getOriginalFilename();
+        return getDirectory() + fileName;
+    }
+
+    public void saveImageFile(final HintFormObject formObject) {
+        if (isFilePresentInHintForm(formObject)) {
```

Comment:
Dám ti tip. Když to napíše takhle:
```
if (!isFilePresentInHintForm(formObject)) {
    return;
}

MultipartFile file = formObject.getImage();
```

Nemusíš mít tak dlouhý if blok a zbavíš se navíc jedné úrovně odsazení pro ten blok kódu. Složitější metody se tak líp čtou.